recipes/search.rb

```diff
@@ -11,6 +12,19 @@
   return
 end
 
-# Keep it simple for now
-node.run_state[cookbook_name]['hosts'] = node[cookbook_name]['hosts']
+if node[cookbook_name]['yggdrasil']['enabled']
+  yggdrasil_host = node[cookbook_name]['yggdrasil']['host']
+  yggdrasil_api_version = node[cookbook_name]['yggdrasil']['api_version']
+  yggdrasil_token = node[cookbook_name]['yggdrasil']['token']
+  yggdrasil_namespace = node[cookbook_name]['yggdrasil']['namespace']
+  yggdrasil_overrides = node[cookbook_name]['yggdrasil']['overrides']
+
+  yggdrasil = Yggdrasil.new(yggdrasil_host,yggdrasil_api_version,yggdrasil_token)
+  yggdrasil_config = yggdrasil.fetch_configs(yggdrasil_namespace,yggdrasil_overrides)
+
+  node.run_state[cookbook_name]['hosts'] = yggdrasil_config["zookeeper_hosts"]
+else
+  node.run_state[cookbook_name]['hosts'] = node[cookbook_name]['hosts']
+end
+
 node.run_state[cookbook_name]['my_id'] = node[cookbook_name]['my_id']
```

Comment:
does yggdrasil store my_id ? (sequentially generated number for identifying member of zookeeper cluster)

Reply:
No, It must be supplied from node attribute